services/speech/src/speechd/connect.js

```diff
@@ -0,0 +1,35 @@
+const net = require("net");
+
+const connectToSocket = require("./connectToSocket");
+const spawnServer = require("./spawnServer");
+const { ConnectionError } = require("./errors");
+
+const connectToSocketOrNull = async params => {
+  try {
+    return await connectToSocket(params);
+  } catch (e) {
+    const isConnectionError = ["ENOENT", "ECONNREFUSED"].includes(e.code);
+    if (isConnectionError) {
+      return null;
```

Comment:
I'm not sure why this returns `null` for a connection error, when it's then conditionally raised later on. Why not have this function throw the connectionError, which can then be optionally caught and suppressed by the function that calls it?

I think that would make this function clearer, and the callee doesn't have to throw it's own exceptions.